bin/test_example.py

```diff
@@ -65,26 +66,24 @@ def attempt_inference(truss_handle, model_version_id, api_key):
         raise Exception(f"Request failed with status code {response.status_code}")
 
 
-def deploy_truss(truss_handle: TrussHandle) -> str:
-    remote_provider = RemoteFactory.create(remote=REMOTE_NAME)
-    model_name = truss_handle.spec.config.model_name
-    for _ in Retrying(
-        wait=wait_random_exponential(multiplier=1, max=120),
-        stop=stop_after_attempt(5),
-        reraise=True,
-    ):
-        service = remote_provider.push(
-            truss_handle, model_name, publish=True, trusted=True
-        )
-        return service.model_version_id
+@retry(
+    wait=wait_random_exponential(multiplier=1, max=120),
+    stop=stop_after_attempt(3),
+    reraise=True,
+)
+def deploy_truss(target_directory: str) -> str:
+    model_deployment = truss.push(
+        target_directory, remote=REMOTE_NAME, trusted=True, publish=True
+    )
+    model_deployment.wait_for_active()
```

Comment:
Nice, so this will block until the model comes up?

Reply:
yep, and fail fast  as soon as there's a failure

Reply:
Could you import this from it's defining module instead? I.e. `from truss.api...`? Forward-compatible with the separation of concerns in the truss package.

Reply:
Wait -- I thought from your PR we were leaving truss.push in the top-level? https://github.com/basetenlabs/truss/pull/1206/files#diff-92d69f0f5363f1d4e214eaf4a10c951888cf1a388ee43f4ed7cd57e720e837e1

There are several users using `truss.push` in their CI jobs now, so removing this would be a pretty serious breaking change (we've documented this API) https://docs.baseten.co/truss-reference/python-sdk

Reply:
I'm going to merge this for now since something seems quite broken with the current script (which calls the legacy predict endpoints), but let's discuss this further, want to make sure we're aligned (and can change if we have to)